### DISCUSSION TITLE
Deselect button when not allowing multiple selection?

### DISCUSSION THREAD

**Discussion by skttl** [leekelleher/umbraco-contentment#435] [Discussion]
When using Buttons as the list editor for Data List and enabling multiple selection, you can deselect a selected button by clicking it again.

If you don't enable multiple selection, you can't do this, but have to relay on the somewhat hidden "Clear selection" feature.

Would you accept a PR that would enable deselecting buttons in a non-multiple selection scenario?

**Comment by leekelleher** [comment]
@skttl Sounds good to me! 👍 (I can't recall why I didn't make it that way to begin with. 🤔)

**Reply by skttl** [reply]
Actually looks like a bug to me, when I look at the source :)

**Reply by leekelleher** [reply]
I now recall my thinking at the time, it was intentional. 😅

My thinking was that in single selection mode, the buttons would act like radio-buttons, where you can select an option, but never de-select it; and for multiple selection mode, it would act like checkbox-list, where options can be toggled on and off.

(For UX perspectives, further reading and insights here: https://ux.stackexchange.com/a/13513)

That said, I do see the value in being able to de-select a single button, after all I did add the "Allow clear?" configuration, _and they aren't radio-buttons_, so I must have considered it at the time. I'll be happy to accept the suggested change. 👍

**Comment by skttl** [comment]
Thats the exact same reasoning I gave the client wanting the feature 😅

**Comment by leekelleher** [comment]
Marking this discussion as closed, thanks for the PRs #436 and #437! 🎉
